src/main/java/appeng/core/worlddata/SpatialDimensionManager.java

```diff
@@ -0,0 +1,223 @@
+
+package appeng.core.worlddata;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilitySerializable;
+import net.minecraftforge.common.util.INBTSerializable;
+
+import appeng.api.storage.ISpatialDimension;
+import appeng.capabilities.Capabilities;
+
+
+public class SpatialDimensionManager implements ISpatialDimension, ICapabilitySerializable<NBTTagCompound>
+{
+	private World world;
+	private Map<Integer, StorageCellData> spatialData = new HashMap<>();
+
+	private static final int MAX_CELL_DIMENSION = 512;
+
+	public SpatialDimensionManager( World world )
+	{
+		this.world = world;
+	}
+
+	@Override
+	public World getWorld()
+	{
+		return this.world;
+	}
+
+	@Override
+	public int createNewCellDimension( Vec3i contentSize, int owner )
+	{
+		int newId = this.getNextId();
+
+		StorageCellData data = new StorageCellData();
+		data.contentDimension = contentSize;
+		data.owner = owner;
+
+		this.initCellArea( data );
+
+		this.spatialData.put( newId, data );
+
+		return newId;
+	}
+
+	@Override
+	public void deleteCellDimension( int cellStorageId )
+	{
+		// TODO cleanup blocks
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)